### DISCUSSION TITLE
How to add a new thermal model to pybamm so that it can be solved with DFN together?

### DISCUSSION THREAD

**Discussion by xyyrly** [pybamm-team/PyBaMM#3070] [Q&A] [answered]
Hi team,

I have another question...How can a user-defined thermal model be added to pybamm and called like：

options = {"thermal": "lumped"}
model = pybamm.lithium_ion.DFN(options)

Thank you very much! Looking forward to a reply :)

**Comment by brosaplanella** [comment]
You should add a submodel in the thermal [submodels folder](https://github.com/pybamm-team/PyBaMM/tree/develop/pybamm/models/submodels/thermal) (note that this requires cloning the repository and installing from source). You can take as an example one of the existing models (e.g. [lumped](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/models/submodels/thermal/lumped.py)), just note that a lot of functionality is inherited from [base_thermal.py](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/models/submodels/thermal/base_thermal.py).

**Reply by xyyrly** [reply]
Yes, I thought about that but I am not familiar with the coding logic.. As I know, base_thermal doesn't include particle domain. If I want to add particle temperature into my thermal model, should I directly define a variable in  def get_fundamental_variables(self):   ?

**Reply by brosaplanella** [reply]
You would have to define the new temperature variables in the new class. Basically, take the example in `lumped.py` and edit your `get_fundamental_variables` method to define the new variables.

**Reply by xyyrly** [reply]
Great! I will try it. Out of curiosity, if I have solved a DFN model, could those parameter values(like concentration) be directly used for the a created thermal model without adding this model as a submodel?

**Reply by brosaplanella** [reply, marked as answer]
You mean solving DFN first and then using the result for the thermal model? It should be possible but not sure what the most efficient approach would be.
